pokemongo_bot/walkers/polyline_walker.py

```diff
@@ -1,13 +1,31 @@
+from geographiclib.geodesic import Geodesic
+
 from pokemongo_bot.walkers.step_walker import StepWalker
 from polyline_generator import PolylineObjectHandler
 
 
 class PolylineWalker(StepWalker):
     def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_lng, dest_alt, speed):
         polyline = PolylineObjectHandler.cached_polyline((self.bot.position[0], self.bot.position[1]), (dest_lat, dest_lng), speed, google_map_api_key=self.bot.config.gmapkey)
-        polyline.set_speed(speed)
 
-        pol_lat, pol_lon = polyline.get_pos()
-        pol_alt = polyline.get_alt() or origin_alt
+        total_distance = Geodesic.WGS84.Inverse(origin_lat, origin_lng, dest_lat, dest_lng)["s12"]
+        distance = min(total_distance, speed)
+        previous_lat, previous_lng, previous_alt = origin_lat, origin_lng, origin_alt
+
+        while True:
+            _, (step_lat, step_lng) = polyline._step_dict[polyline._step_keys[polyline._last_step]]
+            step_alt = polyline.get_alt() or origin_alt
+
+            next_lat, next_lng, next_alt = super(PolylineWalker, self).get_next_position(previous_lat, previous_lng, previous_alt, step_lat, step_lng, step_alt, distance)
+            distance -= Geodesic.WGS84.Inverse(previous_lat, previous_lng, next_lat, next_lng)["s12"]
+            distance = abs(distance)
+
+            if distance > (self.precision + self.epsilon):
```

Comment:
bearing in your case is in the direction of  nextstep but if you have 3 steps within the total distance the final bearing is off.